src/execution_tree/primitives/store_operation.cpp

```diff
@@ -45,31 +51,344 @@ namespace phylanx { namespace execution_tree { namespace primitives
             std::string const& name, std::string const& codename)
       : primitive_component_base(std::move(operands), name, codename, true)
     {}
+    ////////////////////////////////////////////////////////////////////////////////////
+    ///////////////////////////////////////////////////////////////////////////
+    bool store_operation::check_set_parameters(std::int64_t start,
+        std::int64_t stop, std::int64_t step, std::size_t array_length) const
+    {
+        if (start < 0)
+        {
+            start = array_length + start;
+        }
+
+        if (stop < 0)
+        {
+            stop = array_length + stop;
+        }
+
+        if (step > 0)
+        {
+            return (stop > start);
+        }
+        else
+        {
+            return (start > stop);
+        }
+    }
+
+    std::vector<std::int64_t> store_operation::create_list_set(
+        std::int64_t start, std::int64_t stop, std::int64_t step,
+        std::size_t array_length) const
+    {
+        HPX_ASSERT(step != 0);
+        auto actual_start = 0;
+        auto actual_stop = 0;
+
+        if (start >= 0)
+        {
+            actual_start = start;
+        }
+        else    //(start < 0)
+        {
+            actual_start = array_length + start;
+        }
+
+        if (stop >= 0)
+        {
+            actual_stop = stop;
+        }
+        else    //(stop < 0)
+        {
+            actual_stop = array_length + stop;
+        }
+
+        std::vector<std::int64_t> result;
+
+        if (step > 0)
+        {
+            HPX_ASSERT(actual_stop > actual_start);
+            result.reserve((actual_stop - actual_start + step) / step);
+            for (std::int64_t i = actual_start; i < actual_stop; i += step)
+            {
+                result.push_back(i);
+            }
+        }
+        else    //(step < 0)
+        {
+            HPX_ASSERT(actual_start > actual_stop);
+            result.reserve((actual_start - actual_stop - step) / (-step));
+            for (std::int64_t i = actual_start; i > actual_stop; i += step)
+            {
+                result.push_back(i);
+            }
+        }
+
+        if (result.empty())
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
```

Comment:
Is this really an error? I think that slice parameters are allowed to result in empty slices, aren't they?

Reply:
@hkaiser This was leftover from the previous implementation of set. I have removed the error and also simplified the function.